services/index/src/api.js

```diff
@@ -66,6 +66,37 @@ builder.declare({
   return res.reply(helpers.taskUtils.serialize(task));
 });
 
+/** List tasks from given task labels */
+builder.declare({
+  method: 'post',
+  route: '/tasks/indexes',
+  query: paginateResults.query,
+  name: 'findTasksAtIndex',
+  scopes: 'index:list-tasks',
+  input: 'list-tasks-at-index.yml',
+  stability: APIBuilder.stability.experimental,
+  category: 'Index Service',
+  output: 'list-tasks-response.yml',
+  title: 'Find tasks at indexes',
+  description: [
+    'List the tasks given their labels. XXX description',
+  ].join('\n'),
+}, async function (req, res) {
+  // XXX what about authorization?
```

Comment:
Each method has own `scopes` that is being used by api builder to validate if the caller has necessary scopes and he can make that call.

So far I see `index` services is using several scopes:

`index:find-task:<indexPath>` (`findTask`)
`index:list-tasks:<namespace>` (`listTasks`)

where `<indexPath>` and `<namespace>` are parameters that could be passed to the authz step when they are part of the route (i.e. `/task/:indexPath`)
If not, it is necessary to pass those manually via calling `req.authorize({ indexPath: xx })`

In case there will be multiple indexes passed, we would need to construct `scopes` in a way that all of those indexes would be accounted, for example:

```js
scopes: { AllOf: [{
    for: 'indexPath',
    in: 'indexPaths',
    each: 'index:find-task:<indexPath>'
}], }

// with later call:
req.authorize({
    indexPaths: req.body.indexes //  if it's array
});
```

this way we'll make sure that the caller has permissions for all of the tasks he is trying to find. I think it would be simpler to assume initially he needs to have `allOf` those and not to have to deal with partial access.